infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -125,24 +137,17 @@ public SessionManager(@Nonnull TopologyDescriptor topology, CorfuRuntime corfuRu
         this.metadataManager = new LogReplicationMetadataManager(corfuRuntime, topology.getTopologyConfigId());
         this.configManager = new LogReplicationConfigManager(runtime);
         this.upgradeManager = null;
-        replicationContext = new LogReplicationContext(configManager, topology.getTopologyConfigId(), localCorfuEndpoint);
-
-        createSessions();
+        this.routerManager = new LogReplicationRouterManager(topology);
+        this.replicationContext = new LogReplicationContext(configManager, topology.getTopologyConfigId(), localCorfuEndpoint);
     }
 
     /**
      * Refresh sessions based on new topoloogy
      *
      * @param newTopology   the new discovered topology
      */
-    // TODO [V2]: In the connectionModel PR, ensure that refresh is called by the leader node only. Otherwise the
-    //  metadata/status tables will be overwritten by all the nodes in the cluster
     public synchronized void refresh(@Nonnull TopologyDescriptor newTopology) {
 
-        // TODO V2: Make this method a no-op if the new topology has not changed.  Need to override equals and
```

Comment:
can we make this method a no-op if the topology has not changed?

Reply:
I thought about this, and 1 way I could see (and have implemented) is to reduce the number of refresh calls.
If we come here that would mean that atleast the topologyId has changed. And I think we should persist the new topologyId else it can make debugging difficult when something goes wrong.